contracts/StorageMarket_flat.sol

```diff
@@ -2020,48 +2044,49 @@ contract StorageMarketPlace is Sablier {
-
-
-                SellerSubscription[] storage sellerSubscriptions
-             = buyerToSeller[buyer];
-            for (uint256 i = 0; i < buyerSubscriptions.length; i++) {
-                if (buyerSubscriptions[i].buyer == buyer) {
-                    buyerSubscriptions[i].isActive = false;
-                }
-            }
-            for (uint256 i = 0; i < sellerSubscriptions.length; i++) {
-                if (sellerSubscriptions[i].seller == msg.sender) {
-                    sellerSubscriptions[i].isActive = false;
-                }
-            }
-        }
+        // if (stopTime >= now) {
+        //     BuyerSubscription[] storage buyerSubscriptions = sellerToBuyer[msg
+        //         .sender];
+
+
+        //         SellerSubscription[] storage sellerSubscriptions
+        //      = buyerToSeller[buyer];
+        //     for (uint256 i = 0; i < buyerSubscriptions.length; i++) {
+        //         if (buyerSubscriptions[i].buyer == buyer) {
+        //             buyerSubscriptions[i].isActive = false;
+        //         }
+        //     }
+        //     for (uint256 i = 0; i < sellerSubscriptions.length; i++) {
+        //         if (sellerSubscriptions[i].seller == msg.sender) {
+        //             sellerSubscriptions[i].isActive = false;
+        //         }
+        //     }
+        // }
         withdrawFromStream(_streamId, _amount);
+        emit subscription(_streamId);
+
     }
 
-    function cancelSubscription(uint256 _streamId, address seller) public {
+    function cancelSubscription(uint256 _streamId) public {
         // first making the isActive property false in order to filter on UI
-        BuyerSubscription[] storage buyerSubscriptions = sellerToBuyer[seller];
-        SellerSubscription[] storage sellerSubscriptions = buyerToSeller[msg
-            .sender];
-        for (uint256 i = 0; i < buyerSubscriptions.length; i++) {
-            if (buyerSubscriptions[i].buyer == msg.sender) {
-                buyerSubscriptions[i].isActive = false;
-            }
-        }
-        for (uint256 i = 0; i < sellerSubscriptions.length; i++) {
-            if (sellerSubscriptions[i].seller == seller) {
-                sellerSubscriptions[i].isActive = false;
-            }
-        }
+        // BuyerSubscription[] storage buyerSubscriptions = sellerToBuyer[seller];
+        // SellerSubscription[] storage sellerSubscriptions = buyerToSeller[msg
+        //     .sender];
+        // for (uint256 i = 0; i < buyerSubscriptions.length; i++) {
+        //     if (buyerSubscriptions[i].buyer == msg.sender) {
+        //         buyerSubscriptions[i].isActive = false;
+        //     }
+        // }
+        // for (uint256 i = 0; i < sellerSubscriptions.length; i++) {
+        //     if (sellerSubscriptions[i].seller == seller) {
+        //         sellerSubscriptions[i].isActive = false;
+        //     }
+        // }
         cancelStream(_streamId);
+        emit subscription(_streamId);
```

Comment:
Looks like we aren't doing any extra in these new methods, we maybe better off removing these methods?

Reply:
We could maybe actually override the methods and add our own changes or events.